nomad/acl_endpoint.go

```diff
@@ -1593,3 +1611,59 @@ func (a *ACL) GetRoleByName(
 		},
 	})
 }
+
+// policyNamesFromRoleLinks resolves the policy names which are linked via the
+// passed role links. This is useful when you need to understand what polices
+// an ACL token has access to and need to include role links. The function will
+// not return a nil set object, so callers can use this without having to check
+// this.
```

Comment:
Was there a reason for not returning a nil set if we do return a non-nil error? Seems like all callers bail on an err without looking at the set.